src/main/java/org/skriptlang/skript/elements/fireworks/EffElytraBoostConsume.java

```diff
@@ -0,0 +1,90 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package org.skriptlang.skript.elements.fireworks;
+
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import com.destroystokyo.paper.event.player.PlayerElytraBoostEvent;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+
+@Name("Elytra Boost Consume")
+@Description("Change if the firework will be consumed in an <a href='events.html#elytra_boost'>elytra boost event</a>.")
+@Examples({
+	"on player elytra boost:",
+		"\twill consume firework",
+		"\tset to consume the firework"
+})
+@Since("INSERT VERSION")
+public class EffElytraBoostConsume extends Effect {
+
+	static {
+		if (Skript.classExists("com.destroystokyo.paper.event.player.PlayerElytraBoostEvent"))
+			Skript.registerEffect(EffElytraBoostConsume.class, "(set|change) [[the] event] to [:not] consume [the] firework", "[not:do not] consume firework", "set consume firework to %boolean%");
```

Comment:
It's a boolean state. SetEffects pull request isn't merged yet.

This is all grammarly correct, and valid. The boolean expression allows for a variable state to be implemented, which is obviously a positive to the user.

I can add allow and make.

Reply:
If we're talking _grammatically_ correct, then I think `set consumption of firework to false` or `set firework consumption to false` would be much better.

Reply:
Might be a bit farfetched but, what about this. Simpler but might cause confusion with `boolean` as not many people realize it's `yes, on, true` and `no, off, false` 
```applescript
toggle firework consumption [%-boolean%]
```

```
toggle firework consumption on
toggle firework consumption off
toggle firework consumption
```

Reply:
I wouldn't say far fetched but it would be an outlier, similar to [this effect](https://skripthub.net/docs/?id=3024). For that reason, I think that AyhamAl-Ali's suggestion is better with the allow/disallow since there is already an existing effect that uses it, though there is only one. Furthermore, I think that LimeGlass shouldn't really be adding this stuff if his SetEffect PR isn't even guaranteed to be accepted (please correct me if I'm wrong).

Reply:
> I wouldn't say far fetched but it would be an outlier, similar to [this effect](https://skripthub.net/docs/?id=3024). For that reason

Honestly I wouldn't really call that effect a "toggle" as much as allow/disallow, in my uneducated opinion a toggle would be something in relation to intaking no parameter and/or an optional boolean.



> I think that AyhamAl-Ali's suggestion is better with the allow/disallow since there is already an existing effect that uses it

I'm honestly unsure how "allow/disallow" would look for this as `disallow firework consumption` is a weird one
were `enable firework consumption` does work it's basically more limiting than `toggle firework consumption [%-boolean%]` which allows more user control.

> I think that LimeGlass shouldn't really be adding this stuff if his SetEffect PR isn't even guaranteed to be accepted (please correct me if I'm wrong).

The class itself does have it's uses but the `set ... to ...` shouldn't really be included but it's more likely to be accepted as the use cases outweigh the cons.